enhancements/sig-policy/128-resource-selector/README.md

```diff
@@ -0,0 +1,142 @@
+# Resource selector for `ConfigurationPolicy`
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in
+      [website](https://github.com/open-cluster-management-io/open-cluster-management-io.github.io/)
+
+## Summary
+
+Add a resource selector to pair with the existing `namespaceSelector` in order to allow a
+`ConfigurationPolicy` to filter and select objects when an `objectDefinition` doesn't have a name
+provided. The resource selector would also enable the enforcement of `objectDefinitions` without a
+name.
+
+## Motivation
+
+The `ConfigurationPolicy` currently has a `namespaceSelector` to configure the controller to filter
+and range over namespaces relevant to a particular namespaced `objectDefinition`. Furthermore,
+currently an `objectDefinition` without a name is feasible but is not enforceable. A
+`resourceSelector` would both pair well with the `namespaceSelector` for further object filtering as
+well as enable an enforceable unnamed `objectDefinition` by providing the controller with a list of
+names to handle.
+
+Adding a `resourceSelector` would lower the bar to entry for `ConfigurationPolicy`, adding an
+additional configuration available to users without requiring knowledge of Go templating as is
+required with the alternative, `object-templates-raw`.
+
+### Goals
+
+- Add a `resourceSelector` to `ConfigurationPolicy` to allow the controller to handle objects
+  without requiring `object-templates-raw`
+- Make unnamed `objectDefinition`s enforceable
+- Make the `namespaceSelector` apply to `Namespace`s when a `resourceSelector` is not provided for a
+  `Namespace` `objectDefinition`, allowing a related `Namespace` to be created alongside related
+  namespaced objects.
+
+### Non-Goals
+
+- Establishing complex logic in the `resourceSelector` that could not be handled readily by the
+  Kubernetes API and, by extension, the `kubernetes-dependency-watches` library.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As a policy user, I want to be able to create namespaces based on the `namespaceSelector`.
+
+#### Story 2
+
+As a policy user, I want to be able to select objects by label and have the controller iterate over
```

Comment:
Any Kubernetes object can have a label listed in `metadata.labels`. Currently if an `objectDefinition` provides a definition without `metadata.name`, the controller will handle it but will refuse to enforce it. With the `resourceSelector`, it's being fed a list of names, so we can enable enforcement since it's a user-defined filtered list.

Reply:
"enforce" We usually create new resources. but how `resourceSelector` apply to this?

Reply:
This wouldn't apply to creating new resources. This would only apply to objects that already exist.